### DISCUSSION TITLE
Allow migrate dev to skip generating a new migration

### DISCUSSION THREAD

**Discussion by luxaritas** [prisma/prisma#19900] [Q&A]
## Problem
This is a "part 2" to #19894. I want to enable a development environment where on app start, migrations are automatically applied. I would do this during a docker compose up. However, if the user has modified the schema.prisma, this would cause a new migration to be generated, which is not the desired behavior here.

## Suggested solution
Add a `--skip-create` flag to skip creation of a migration if once all migrations are applied, there are further differences between the current state and the prisma schema

## Alternatives

* `db push` Push doesn't run data migrations, which is required in my use case - an extra db seed would be required which would mean the data would have to be duplicated in the seed script AND seeding would be forced to be idempotent as seeding would have to be run on every app start
* Don't automatically migrate
* Manually remove the generated migration

## Additional context
This appears to be a straightforward change, and I'd be happy to file a PR. Please let me know if a PR would be accepted!

**Comment by janpio** [comment]
Are you looking for `migrate deploy`: https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-deploy?

**Reply by luxaritas** [reply]
No - I still want to seed, generate, and reset the database if necessary

**Reply by luxaritas** [reply]
@janpio Given the above, is it appropriate to re-open this as a feature request? Or is there some other way you'd figure my use case should be handled aside from extending `migrate dev`? (Allow `migrate deploy` to reset and seed? Though at that point, you're adding functionality that should really only happen in dev)

**Reply by janpio** [reply]
I had understood you already have data in the database, so an automated `db seed` would not really make sense - but otherwise, I would suggest to just run the command. Same for `prisma generate`. 

A reset is not really something that you would want to run automatically, would it? And if you know you want one, then we have it as a standalone command for it: https://www.prisma.io/docs/reference/api-reference/command-reference#migrate-reset That could also replace `migrate deploy`.

The main job of `migrate dev` is to create migrations. Adding a flag to remove that would certainly be possible, but would lead to a really weird mental model what the command is about.

**Reply by luxaritas** [reply]
> A reset is not really something that you would want to run automatically, would it? And if you know you want one, then we have it as a standalone command for it

The intention here is to simplify development environment setup as much as possible. I don't want to reset the DB _every_ time, only when a reset is needed. Data in development is almost certainly going to be without value anyways, so if I've eg switched branches with a different set of migrations, when I start up my environment I imagine it should just do what it needs to do to make the environment match. Even if it does prompt to reset if needed (which I think would cause problems since it's running under Docker Compose with other services being started up), I still don't think any existing commands handles the use case (migrate dev potentially creates migrations, migrate deploy never resets or seeds if needed, and db push won't include my data migrations so I'd need to define those twice, both in my migrations and my seed script)

**Reply by luxaritas** [reply]
I suppose one benefit of db push + db seed is that if I change my seed data it's always reapplied, and there's an additional level of control for if the data migrations don't completely match what would be expected in a dev environment. But I'm still not super happy that some piece of data that has to be in a master table could be messed up because the way the data gets there in dev vs prod is different

**Reply by janpio** [reply]
Isn't then `migrate reset` the tool you are looking for?
(The problem with using `db push` is that your migration history table then is not in the right state, and you can not use `migrate dev` to create new migrations at all before fixing that)

**Reply by luxaritas** [reply]
Ooh was not aware of that footgun with DB push.

The reason I don't want to do migrate reset is that I don't want to reset development data _unless I have to_. If I added a new test user, stop the service, then restart the service, there's no reason why I should have to reset the DB because the schema didn't change. Similarly if I can just apply any unapplied migrations without issue, there's no reason I should need to reset my testing data.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
